runtime/src/main/java/org/corfudb/runtime/collections/PersistedStreamingMap.java

```diff
@@ -0,0 +1,320 @@
+            // If the iterator was created by some thread, it also has to be consumed by it.
+            if (!iterator.isOwningHandle()) {
+                throw new IllegalStateException("Detected multi-threaded access to this iterator.");
+            }
+
+            try {
+                iterator.status();
+            } catch (RocksDBException e) {
+                throw new UnrecoverableCorfuError(
+                        "There was an error reading the persisted map.", e);
+            }
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public boolean hasNext() {
+            // If we have the next element pipelined, go ahead and return true.
+            if (next != null) {
+                return true;
+            }
+
+            // If the iterator is valid, this means that the next entry exists.
+            checkInvariants();
+            if (iterator.isValid()) {
+                // Go ahead and cache that entry.
+                next = new AbstractMap.SimpleEntry(
+                        serializer.deserialize(Unpooled.wrappedBuffer(iterator.key()), corfuRuntime),
+                        serializer.deserialize(Unpooled.wrappedBuffer(iterator.value()), corfuRuntime));
+                // Advance the underlying iterator.
+                iterator.next();
+            } else {
+                // If there is no more elements to consume, we should release the resources.
+                iterator.close();
+            }
+
+            return next != null;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public Entry<K, V> next() {
+            if (hasNext()) {
+                current = next;
+                next = null;
+                return current;
+            } else {
+                throw new NoSuchElementException();
+            }
+
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int size() {
+        return dataSetSize.get();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean isEmpty() {
+        return dataSetSize.get() == 0;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean containsKey(@NonNull Object key) {
+        final ByteBuf keyPayload = Unpooled.buffer();
+        serializer.serialize(key, keyPayload);
+        try {
+            byte[] value = rocksDb.get(
+                    keyPayload.array(), keyPayload.arrayOffset(), keyPayload.readableBytes());
+            return value != null;
+        } catch (RocksDBException ex) {
+            throw new UnrecoverableCorfuError(ex);
+        } finally {
+            keyPayload.release();
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * Please use {@link StreamingMap#entryStream()}.
+     */
+    @Override
+    public boolean containsValue(@NonNull Object value) {
+        throw new UnsupportedOperationException();
```

Comment:
if I'm not wrong rocks provides the opportunity to check that the value exists in the db
why not add that here?